grammar.js

```diff
@@ -270,8 +273,36 @@ module.exports = grammar({
       field('rhs', $._expression)
     )),
 
-    _operator: $ => choice( "+", "-", "*", "/", "%", "&", "|", "^", "**", ">>", "<<", "==", "!=", "<", "<=", ">", ">=", "<=>", "===", "[]", "[]?", "[]=", "!", "~", "!~", "=~",),
+    /**
+     * @see {@link https://crystal-lang.org/reference/syntax_and_semantics/classes_and_methods.html}
+     * @see {@link https://crystal-lang.org/reference/syntax_and_semantics/generics.html}
+     */
+    class_definition: $ => {
+      return seq(
+        'class',
+        field('name', $.type),
+        optional(seq('<', field('superclass', $.type))),
+        repeat($._statement),
+        'end'
+      );
+    },
 
+    type: $ => seq(
+      choice(
+        alias($.constant, 'type'), // no namespace
```

Comment:
This effectively "inlines" this bit, causing a result to be `(type)` instead of `(type (constant))`